workers/common/src/main/kotlin/common/context/WorkerContextImpl.kt

```diff
@@ -114,6 +115,16 @@ internal class WorkerContextImpl(
             c.mapValues { entry -> entry.value.resolveSecrets(resolvedSecrets) }
         }.orEmpty()
 
+    override suspend fun resolveProviderPluginConfigSecrets(
+        config: List<ProviderPluginConfiguration>?
+    ): List<ProviderPluginConfiguration> =
+        config?.let { c ->
+            val secrets = c.flatMap { it.secrets.values }
+            val resolvedSecrets = parallelTransform(secrets, configSecretsCache, this::resolveConfigSecret) { it }
+
+            c.map { it.resolveSecrets(resolvedSecrets) }
```

Comment:
The function is basically a copy of the function above just for another datatype. Any specific suggestions?

Reply:
Ok, if there is another extension function with this name, then better keep it. To increase readability, could then a named lambda parameter be used instead of `it`? `pluginConfig.resolveSecrets(...)` would be less ambiguous.